doc/lib.rst

```diff
@@ -66,7 +66,7 @@ Core
 
 * `macrocache <macrocache.html>`_
   Provides an API for macros to collect compile-time information across modules.
-  
+
```

Comment:
> removed mersenne and sums because they're deprecated (assuming Deprecate std/sums #18439 gets merged)

instead, the removal of sums from lib.rst should be done in #18439; for mersenne it's fine since that module was already deprecated